docs/source/api/apollo-gateway.mdx

```diff
@@ -34,9 +37,14 @@ The core of the federated gateway implementation. For an example, see the ["impl
 
   * `buildService`: `(service: ServiceDefinition) => GraphQLDataSource`
 
-    A function to be called for each service in `options.serviceList`.
-    The `service` is provided as an argument, and a `GraphQLDataSource` is expected to be returned.
-    Allows for customizing transports.
+    Define this function to customize your gateway's data transport to some or
+    all of your data graph's implementing services.
+
+    If you provide this function, `ApolloGateway` calls it for every
+    implementing service in your data graph. The function must return a `GraphQLDataSource` object.
+
+    The example below demonstrates adding an `x-user-id`
```

Comment:
It might be worth pointing out specifically that this can support non-HTTP protocols 🤷‍♂